packages/jsapi-shim/src/dh.types.ts

```diff
@@ -0,0 +1,682 @@
+   * @param max the optional max value visible on this axis. If min is specified, max is also expected.
+   */
+  range(pixelCount?: number, min?: any, max?: any): void;
+}
+
+export interface SeriesDataSource {
+  readonly axis: Axis;
+  readonly type: SourceType;
+  readonly columnName: string;
+}
+
+export interface OneClick {
+  readonly columns: { name: string; type: string }[];
+  readonly isRequireAllFiltersToDisplay: boolean;
+
+  setValueForColumn(columnName: string, value: any): void;
+  getValueForColumn(columnName: string): any;
+}
+
+export interface Column {
+  readonly type: string;
+  readonly name: string;
+  readonly description: string;
+  readonly constituentType: string;
+
+  readonly isPartitionColumn: boolean;
+
+  filter(): FilterValue;
+  sort(): Sort;
+}
+export interface FilterValue {
+  ofString(input: object): FilterValue;
+  ofNumber(input: number): FilterValue;
+  ofNumber(input: LongWrapper): FilterValue;
+  ofNumber(input: object): FilterValue;
+  ofBoolean(input: boolean): FilterValue;
+
+  eq(value: FilterValue): FilterCondition;
+  eqIgnoreCase(value: FilterValue): FilterCondition;
+  notEq(value: FilterValue): FilterCondition;
+  notEqIgnoreCase(value: FilterValue): FilterCondition;
+  greaterThan(value: FilterValue): FilterCondition;
+  lessThan(value: FilterValue): FilterCondition;
+  greaterThanOrEqualTo(value: FilterValue): FilterCondition;
+  lessThanOrEqualTo(value: FilterValue): FilterCondition;
+  inIgnoreCase(value: FilterValue): FilterCondition;
+  notIn(value: FilterValue): FilterCondition;
+  notInIgnoreCase(value: FilterValue): FilterCondition;
+  contains(value: FilterValue): FilterCondition;
+  isFalse(): FilterCondition;
+  isTrue(): FilterCondition;
+  isNull(): FilterCondition;
+  invoke(method: string, ...args: FilterValue[]): FilterCondition;
+}
+export interface FilterCondition {
+  invoke(method: string, ...args: FilterValue[]): FilterCondition;
+  search(value: FilterValue, columns?: FilterValue[]): FilterCondition;
+
+  not(): FilterCondition;
+  and(first: FilterCondition, ...rest: FilterCondition[]): FilterCondition;
+  or(first: FilterCondition, ...rest: FilterCondition[]): FilterCondition;
+
+  toString(): string;
+  // columns: Column[] //doesnt work
+}
+export interface Sort {
+  reverse(): Sort;
+
+  readonly column: Column;
+  readonly direction: string;
+
+  readonly isAbs: boolean;
+
+  asc(): Sort;
+  desc(): Sort;
+  abs(): Sort;
+}
+export interface Table extends Evented {
+  readonly EVENT_SIZECHANGED: string;
+  readonly EVENT_UPDATED: string;
+  readonly EVENT_ROWADDED: string;
+  readonly EVENT_ROWREMOVED: string;
+  readonly EVENT_ROWUPDATED: string;
+  readonly EVENT_SORTCHANGED: string;
+  readonly EVENT_FILTERCHANGED: string;
+  readonly EVENT_CUSTOMCOLUMNSCHANGED: string;
+  readonly EVENT_DISCONNECT: string;
+  readonly EVENT_RECONNECT: string;
+  readonly EVENT_RECONNECTFAILED: string;
+
+  readonly SIZE_UNCOALESCED: number;
+
+  reverse(): Sort;
```

Comment:
Kinda weird this is in the middle of all the props/readonly stuff, should just have it after